File: Tugas Besar/Prodi.py
import requests
import json
class Prodi:
    def __init__(self):
        self.__id=None
        self.__kode_prodi = None
        self.__nama_prodi = None
        self.__url = "http://a0832601.xsph.ru/AppTugasBesar/prodi_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def kode_prodi(self):
        return self.__kode_prodi
        
    @kode_prodi.setter
    def kode_prodi(self, value):
        self.__kode_prodi = value
    @property
    def nama_prodi(self):
        return self.__nama_prodi
        
    @nama_prodi.setter
    def nama_prodi(self, value):
        self.__nama_prodi = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_kode_prodi(self, kode_prodi):
        url = self.__url+"?kode_prodi="+kode_prodi
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id_prodi']
            self.__kode_prodi = item['kode_prodi']
            self.__nama_prodi = item['nama_prodi']
        return data
    def simpan(self):
        payload = {
            "kode_prodi":self.__kode_prodi,
            "nama_prodi":self.__nama_prodi
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_kode_prodi(self, kode_prodi):
        url = self.__url+"?kode_prodi="+kode_prodi
        payload = {
            "kode_prodi":self.__kode_prodi,
            "nama_prodi":self.__nama_prodi
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_kode_prodi(self,kode_prodi):
        url = self.__url+"?kode_prodi="+kode_prodi
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text

File: Tugas Besar/Prodi.php
<?php
//Simpanlah dengan nama file : Prodi.php
require_once 'database.php';
class Prodi 
{
    private $db;
    private $table = 'prodi';
    public $kode_prodi = "";
    public $nama_prodi = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_kode_prodi(int $kode_prodi)
    {
        $query = "SELECT * FROM $this->table WHERE kode_prodi = $kode_prodi";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`kode_prodi`,`nama_prodi`) VALUES ('$this->kode_prodi','$this->nama_prodi')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET kode_prodi = '$this->kode_prodi', nama_prodi = '$this->nama_prodi' 
        WHERE id_prodi = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_kode_prodi($kode_prodi): int
    {
        $query = "UPDATE $this->table SET kode_prodi = '$this->kode_prodi', nama_prodi = '$this->nama_prodi' 
        WHERE kode_prodi = $kode_prodi";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id_prodi = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_kode_prodi($kode_prodi): int
    {
        $query = "DELETE FROM $this->table WHERE kode_prodi = $kode_prodi";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>